src/components/share-overlay/share-overlay.js

```diff
@@ -535,7 +538,11 @@ class ShareOverlay extends Component {
     let url = this.state.embedUrl;
     const template = this.state.embedTemplate;
     url = this._maybeAddParamsToUrl(url);
-    return template.replace(/{embedUrl}/g, url);
+    const {width, height} = this.props.player.getPlayerData();
+    const toReplace = {['{embedUrl}']: url, ['{width}']: width || PLAYER_WIDTH_EMBED_DEFAULT, ['{height}']: height || PLAYER_HEIGHT_EMBED_DEFAULT};
```

Comment:
we already have a code in the player that replaces configuration template values for share, actually. 
https://github.com/kaltura/kaltura-player-js/blob/5cb90f12b89d82606ea0136509fd2795d1acf57d/src/common/plugins/plugins-config-store.ts#L51

you can replace the template values with the real ones on the player side and then just use them here.